src/GenesisMakeFileWindow.cpp

```diff
@@ -137,21 +158,67 @@ void GenesisMakeDirWindow::MessageReceived(BMessage* message)
 }
 
 ////////////////////////////////////////////////////////////////////////
-bool GenesisMakeDirWindow::CreateFolder(const char *dirpath, const char *dirname)
+bool GenesisMakeFileWindow::CreateFolder(const char *dirpath, const char *dirname)
 ////////////////////////////////////////////////////////////////////////
 {
 	BString dir;
 	dir.SetTo(dirpath);
 	dir+="/";
 	dir+=dirname;
 
-	if (create_directory(dir.String(), 0777)==B_OK)		// TODO: jo a 0777?
+	BEntry *entry = new BEntry(dir);
+	if (entry->Exists() && entry->IsDirectory())
+	{
+		BAlert *alert = new BAlert("Cannot create new folder","Folder already exists","OK",NULL,NULL,B_WIDTH_AS_USUAL,B_INFO_ALERT);
+		alert->Go();
 		return 1;
-	else
-		return 0;
+	}
+	
+	status_t status = create_directory(dir.String(), 0777);
+	if (status == B_OK)
+		return 1;
+		
+	BString errormsg;
+	switch (status){
+		case B_NOT_A_DIRECTORY:
+			errormsg.SetToFormat("Entry '%s' already exists", dirname);
```

Comment:
Maybe a bit too technical. As this alert only appears when it's not a folder, we may be more specific, like: _"A file '%s' already exists."_